internal/core/src/index/VectorMemIndex.cpp

```diff
@@ -112,6 +171,101 @@ VectorMemIndex<T>::Load(const BinarySet& binary_set, const Config& config) {
     LoadWithoutAssemble(binary_set, config);
 }
 
+template <typename T>
+void
+VectorMemIndex<T>::LoadV2(const Config& config) {
+    if (config.contains(kMmapFilepath)) {
+        return LoadFromFileV2(config);
+    }
+
+    auto blobs = space_->StatisticsBlobs();
+    std::unordered_set<std::string> pending_index_files(blobs.size());
+    for (auto& blob : blobs) {
+        pending_index_files.insert(blob.name);
+    }
+
+    auto res = space_->GetBlobByteSize(std::string(INDEX_FILE_SLICE_META));
+    std::map<std::string, storage::FieldDataPtr> index_datas{};
+
+    if (!res.ok() && !res.status().IsFileNotFound()) {
+        PanicInfo(DataFormatBroken, "failed to read blob");
+    }
+    bool slice_meta_exist = res.ok();
+
+    auto read_blob = [&](const std::string& file_name)
+        -> std::unique_ptr<storage::DataCodec> {
+        auto res = space_->GetBlobByteSize(file_name);
+        if (!res.ok()) {
+            PanicInfo(DataFormatBroken, "unable to read index blob");
+        }
+        auto index_blob_data =
+            std::shared_ptr<uint8_t[]>(new uint8_t[res.value()]);
+        auto status = space_->ReadBlob(file_name, index_blob_data.get());
+        if (!status.ok()) {
+            PanicInfo(DataFormatBroken, "unable to read index blob");
+        }
+        return storage::DeserializeFileData(index_blob_data, res.value());
+    };
+    if (slice_meta_exist) {
+        pending_index_files.erase(INDEX_FILE_SLICE_META);
+        auto slice_meta_sz = res.value();
+        auto slice_meta_data =
+            std::shared_ptr<uint8_t[]>(new uint8_t[slice_meta_sz]);
+        auto status =
+            space_->ReadBlob(INDEX_FILE_SLICE_META, slice_meta_data.get());
+        if (!status.ok()) {
+            PanicInfo(DataFormatBroken, "unable to read slice meta");
+        }
+        auto raw_slice_meta =
+            storage::DeserializeFileData(slice_meta_data, slice_meta_sz);
+        Config meta_data = Config::parse(std::string(
+            static_cast<const char*>(raw_slice_meta->GetFieldData()->Data()),
+            raw_slice_meta->GetFieldData()->Size()));
+        for (auto& item : meta_data[META]) {
+            std::string prefix = item[NAME];
```

Comment:
what about use `std::string_view`

Reply:
> what about use `�std::string_view`

prefix will be used as string& in a function later, but string_view can not be cast to string implicitly